Tool/SuperCAT.ps1

```diff
@@ -1,398 +1,359 @@
+        [String]$RootDirectory
+    )
 
-    if($System -eq 1){
-        Add-Content -value "System: $System1" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($System -eq 2){
-        Add-Content -value "System: $System2" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($System -eq 3){
-        Add-Content -value "System: $System3" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($System -eq 4){
-        Add-Content -value "System: $System4" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($System -eq 5){
-        Add-Content -value "System: $System5" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($Location -eq 6){
-        Write-Output "Type in the System and press Enter"
-        $Location6 = Read-Host
-        Add-Content -value "Location: $System6" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
+    function Read-SystemType {
+        param (
+            [Parameter(Mandatory=$True,Position=1)]
+            [ValidateNotNullOrEmpty()]
+            [String]$RootDirectory
+        )
 
-    ## Based on what is in the JSON file, it asks for the location of the machine (or you can write it in).
-    Write-Output "
-    ===============================================================
-    Where is this system located?
-    1 = $Location1
-    2 = $Location2
-    3 = $Location3
-    4 = $Location4
-    5 = $Location5
-    6 = Other (I will type it in)
-    ==============================================================="
-
-    if($PSBoundParameters.ContainsKey("Location")){
-        $Location = $PSBoundParameters.Location
-    }
-    else{
-        $Location = Read-Host
-    }
+        ## Autopull if available
+        if (Test-Path -Path "C:\Capre_rel.cer" -PathType Leaf) {
+            return "CAPRE"
+        }
 
-    while($System -notmatch "^[1-6]$"){
-        $Location = Read-Host "You must input a number between 1 and 6."
-    }
+        ## Request input if not
+        Try {
+            $PlatformList = Import-CSV -Path "$RootDirectory\PlatformList.csv"
+        } Catch {
+            throw "PlatformList.csv doesn't exist! Please reinstall SuperCAT."
+        }
 
-    if($Location -eq 1){
-        Add-Content -value "Location: $Location1" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($Location -eq 2){
-        Add-Content -value "Location: $Location2" -Path "$GatherLogs\$ComputerName-Info.txt"
-    }
-    elseif($Location -eq 3){
-        Add-Content -value "Location: $Location3" -Path "$GatherLogs\$ComputerName-Info.txt"
+        $result = $(Read-Intent $PlatformList.SystemName "What type of system is this?")
+        if ( $result -eq "Other") {
+            return $(Read-Host -Prompt "What type of system is this")
+        } else {
+            return $result
+        }
     }
-    elseif($Location -eq 4){
-        Add-Content -value "Location: $Location4" -Path "$GatherLogs\$ComputerName-Info.txt"
+    function Read-SystemVersion {

```

Comment:
Need Information on what this is. MajorMinor of what? System's utilize CPINs for versioning of software.

Reply:
CPINs, being between 17 and 39 characters long are impractical to stuff into a file name.  Do we feel that having quick reference as to the version of any part of the system is valuable when performing system analysis? I'm not tied to the MajorMinor format, or even the existence of the field.

Reply:
I agree CPIN is too long, however, I'm not sure what most users would put here.  Do the systems we're looking at prototyping this on have a Major/Minor label?

Reply:
I known the system we have the dev build for does, but I have no knowledge of the other systems. I would expect them to though, as it's the standard model for software versioning (Major.Minor.Bugfix rRelease)

Reply:
We can leave it for now... I'd rather address a file naming solution.  In addition, in the future, I think some of these user entered details should make their way into the -info.csv file instead.